great_expectations/core/batch.py

```diff
@@ -726,7 +726,10 @@ def get_batch_request_dict(
     batch_request: Optional[BatchRequest] = None, validations: Optional[list] = None
 ) -> dict:
     if isinstance(batch_request, BatchRequest):
-        if batch_request.runtime_parameters.get("batch_data") is not None:
+        if (
```

Comment:
If `batch_request` is `BatchRequest` and not `RuntimeBatchRequest`, we need to also check for `runtime_parameters`.